src/main.cpp

```diff
@@ -1498,7 +1495,7 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
         // We don't yet know if the transaction commits to consensusBranchId,
         // but if the entry gets added to the mempool, then it has passed
         // ContextualCheckInputs and therefore this is correct.
-        CTxMemPoolEntry entry(tx, nFees, GetTime(), dPriority, chainActive.Height(), mempool.HasNoInputsOf(tx), fSpendsCoinbase, consensusBranchId);
+        CTxMemPoolEntry entry(tx, nFees + nFeeDelta, GetTime(), chainActive.Height(), mempool.HasNoInputsOf(tx), fSpendsCoinbase, consensusBranchId);
```

Comment:
I can't see where the delta was being applied to `nFees` before here previously, so I think the `+ nFeeDelta` change here is either an error, or fixing a bug that isn't apparently obvious from the commit history.

Reply:
previously, `ApplyDeltas` was adding `feeDelta` to `nModifiedFees` (which we initialise as `nFees`):

```c++
void CTxMemPool::ApplyDeltas(const uint256 hash, double &dPriorityDelta, CAmount &nFeeDelta)
{
    LOCK(cs);
    std::map<uint256, std::pair<double, CAmount> >::iterator pos = mapDeltas.find(hash);
    if (pos == mapDeltas.end())
        return;
    const std::pair<double, CAmount> &deltas = pos->second;
    dPriorityDelta += deltas.first;
    nFeeDelta += deltas.second;
}

```
https://github.com/zcash/zcash/blob/master/src/txmempool.cpp#L703

Reply:
this doesn't show up in the commit history because it was overwritten in force-pushes. i'll do another force-push showing the diff from the older version.